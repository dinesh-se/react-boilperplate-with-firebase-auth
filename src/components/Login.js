import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth'

export const Login = ({ startLogin }) => (
  <div className="box-layout">
    <section className="box-layout__box">
      <h1 className="box-layout__title">App Title</h1>
      <p>Tagline for app</p>
      <button
        className="button"
        onClick={startLogin}
      >
        Login with Google
      </button>
    </section>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(Login);