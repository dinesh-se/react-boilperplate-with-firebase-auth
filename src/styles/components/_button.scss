.button {
  background: $blue;
  border: none;
  display: inline-block;
  font-weight: 300;
  padding: $s-size;
  color: #ffffff;
  font-size: $font-size-large;
  text-decoration: none;
  line-height: 1;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}