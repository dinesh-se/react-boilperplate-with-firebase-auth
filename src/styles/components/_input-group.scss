.input-group {
  display: flex;
  flex-direction: column;
  margin-bottom: $m-size;
  @media (min-width: $desktop-transition-point){
    flex-direction: row;
    margin-bottom: $l-size;
  }
}

.input-group__item {
  margin-bottom: $s-size;

  @media (min-width: $desktop-transition-point) {
    margin: 0 $s-size 0 0;
  }
}