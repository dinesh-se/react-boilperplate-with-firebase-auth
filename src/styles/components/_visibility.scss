.show-for-mobile {
  @media (min-width: $desktop-transition-point) {
    display: none;
  }
}

.show-for-desktop {
  @media (max-width: $desktop-transition-point - 0.1rem) {
    display: none;
  }
}