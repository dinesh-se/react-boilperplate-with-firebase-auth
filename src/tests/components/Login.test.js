import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../../components/Login';

let startLogin, wrapper;

beforeEach(() => {
  startLogin = jest.fn();
  wrapper = shallow(<Login startLogin={startLogin} />);
});

test('should render Header correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call start login on button click', () => {
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});