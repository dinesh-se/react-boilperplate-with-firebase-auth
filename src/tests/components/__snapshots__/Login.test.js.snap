// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div
  className="box-layout"
>
  <section
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      App Title
    </h1>
    <p>
      Tagline for app
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </section>
</div>
`;
